### ISSUE TITLE
Duplicate class in androidx.annotation

### ISSUE THREAD

**Issue by Nek-12** [Lavmee/constraintlayout-compose-multiplatform#61]
Build fails with: 

```
Duplicate class androidx.annotation.AnimRes found in modules annotation-desktop-1.8.0-alpha01.jar -> annotation-desktop-1.8.0-alpha01 (org.jetbrains.compose.annotation-internal:annotation-desktop:1.8.0-alpha01) and annotation-jvm-1.8.1.jar -> annotation-jvm-1.8.1 (androidx.annotation:annotation-jvm:1.8.1)
```

Usage:
```kotlin
commonMain.dependencies {
    api(libs.compose.constraintlayout)
}
```

Other modules use other androidx. libraries. All modules use only this artifact for constraint layout. transitive dependencies may use the original artifact (unknown)

Workaround:
See comments below.

**Comment by Lavmee** [comment]
At the moment, the internal JetBrains library cannot be replaced with the official one, since there is no JS target there.

**Comment by Lavmee** [comment]
@Monabr, I'll check everything this evening.

**Comment by Nek-12** [comment]
I added this dependency 3 times. In iOSMain (and other targets), I added it without the exclude. In android main, I added it with exclude. In commonMain, I added it with compileOnly. Then I declared it as compileOnly in every module that uses code from constraint layout in commonMain. That configuration works, but I agree that it is extremely frustrating and just dumb that the author included a duplicated class without testing the functionality, and this needs to be fixed on the lib level.
